components/api/idp/validator.bal

```diff
@@ -0,0 +1,87 @@
+// ------------------------------------------------------------------------
+//
+// Copyright 2019 WSO2, Inc. (http://wso2.com)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+//
+// ------------------------------------------------------------------------
+
+import ballerina/config;
+import ballerina/http;
+import ballerina/log;
+import ballerina/time;
+import ballerina/io;
+
+boolean isIntrospectionEPInitialized = false;
+
+public type TokenDetail record { 
+    string username;
+    int expiryTime;
+};
+
+// http:Client introspectionEP3 = new(config:getAsString(constants:IDP_INTROSPCET_VAR));
```

Comment:
Remove this commented line if it is not required. If this temporary, add a todo comment